### ISSUE TITLE
all MW-score for each function is 2.3 in the MW-score_result.txt

### ISSUE THREAD

**Issue by B-1991-ing** [AnantharamanLab/METABOLIC#124]
Hi Zhichao,

I took the assembled contig and the metagenomic reads as the input of the METABOLIC-C.pl. But in the MW-score_result/MW-score_result.txt, I found that **MW-score for each function** of all functions are 2.3. Is there something wrong?

[MW-score_result.txt](https://github.com/AnantharamanLab/METABOLIC/files/10674568/MW-score_result.txt)

Best,

Bing

**Comment by ChaoLab** [comment]
Hi,
If the input genome is only one (the assembled contig), the MW-score will be 100 / (no. of all metabolic steps). In this case, it will be 100 / 43 ~= 2.3. So each step's MW-score here is not meaningful now.

**Comment by B-1991-ing** [comment]
Thank you for your reply, Zhichao. 

Could you give me an equation to exactly **calculate the MW-score of each function**? 

Can I directly see **the gene depth of genes responsible for each function** (of the total 43 functions)?

**Comment by ChaoLab** [comment]
You will see a gene depth (coverage) txt file in the resulting METABOLIC output folder.
The equations to get MW-score can be found in our Microbiome paper: https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-021-01213-8

**Comment by B-1991-ing** [comment]
Yeah, I saw the All_gene_collections_mapped.depth.txt file generated in the output folder. But, I want to use the exact same 43 functional categories as summarised in the MW-score_result.txt. So, the 43 functional categories were classified from **which sheet of the METABOLIC_result.xlsx** --- HMMHitNum, FunctionHit, KEGGModuleHit, KEGGModuleStepHit, dbCAN2Hit, MEROPSHit?

**Comment by B-1991-ing** [comment]
Hi Zhichao,

Could you briefly answer my question above? 

Best,

Bing

**Comment by ChaoLab** [comment]
> Yeah, I saw the All_gene_collections_mapped.depth.txt file generated in the output folder. Are the 43 MW function included in the file --- /METABOLIC/METABOLIC_template_and_database/MW-score_reaction_table.txt (53 rows in total)?
> 
> All_gene_collections_mapped.depth.txt shows the total average depth of each gene and **not the the total average depth of each MW function**. But the HMMHitNum sheet of the METABOLIC_result.xlsx shows the hit numbers of each MW function, could I use the the hit numbers of each MW function to divide the total metagenomic reads number to get a MW function gene depth value (or MW function coverage)? Because the total hit number of each function were found from each metagenomic reads file?
> 
> MW function coverage = the hit numbers of each MW function / the total metagenomic reads number of each sample [METABOLIC_result.xlsx](https://github.com/AnantharamanLab/METABOLIC/files/10690144/METABOLIC_result.xlsx)

Yes to the first question.
The total hit number of each function is calculated according to the functional annotation of the input metagenomic assembly (for your case). It is not related to reads

**Comment by B-1991-ing** [comment]
Okay, if I want to see each unction depth within each contig file, I can do as below?

MW function coverage = the hit numbers of each MW function / the total contig sequence number of each sample

Here one sample has one assembled contig file, so **the total contig sequence number of each sample** is directly calculated from each contig file.
